client/state/selectors/is-user-registration-days-within-range.js

```diff
@@ -0,0 +1,31 @@
+/**
+ * External dependencies
+ */
+import i18n from 'i18n-calypso';
+
+/**
+ * Internal dependencies
+ */
+import { getCurrentUserDate } from 'state/current-user/selectors';
+
+/**
+ * Returns true if the the number of days the current user has been registered for falls within the specied range of values.
+ *
+ * @param  {Object}  state  Global state tree
+ * @param  {Number}  from   Lower bound on days
+ * @param  {Number}  to     Upper bound on days
+ * @return {bool}           True if the number of days falls withing the specified range
+ */
+const isUserRegistrationDaysWithinRange = ( state, from, to ) => {
+	const date = getCurrentUserDate( state );
+
+	if ( ! date ) {
+		return false;
+	}
+
+	const days = i18n.moment().diff( date, 'days', true );
```

Comment:
It looks like this is calculating the difference in days between the current time and the user date, so the selector would return different results for the same inputs depending on the date.

Generally we try to keep selectors as pure functions, which comes with benefits like memoization and makes testing much simpler. In this case we could accept the moment instance as a function parameter, eg:

```js
const isUserRegistrationDaysWithinRange = ( state, moment, from, to ) => {
  // ...
  const days = moment.diff( date, 'days', true );
  // ..
};
```